### PR TITLE
Search features

### PR DESCRIPTION
**Pull request by mbooker5**
Description
This PR addresses feedback from the last PR
This PR implements two search filters for activities within a specified distance and for activities that contain at least one of the selected categories

Features Added
User can filter activities by distance from their current location
User can filter activities by selecting categories that they want the activities to have
User can utilize both filters simultaneously

Test Plan
// I was rate limited by Parse while making the test plan

### CODE REVIEW

File: Urban-Fun/View Controllers/FiltersViewController.m [hunk 1]
```diff
@@ -7,25 +7,128 @@
...
+    }
+    [self setDistanceSliderLabel];
+    [self setSelectCategoriesLabel];
+}
+
+- (IBAction)distanceChanged:(id)sender {
+    float sliderValue = floorf(self.distanceSlider.value);
+    NSLog(@"%f", sliderValue);
+    
+    if (sliderValue == 1){
```
[the remark this answers is not in the document]
Comment (wxie-git) [reply Urban-Fun/View Controllers/FiltersViewController.m:36]: Objc has some nice syntax sugar for creating things like NSNumber and NSString, eg to create a NSNumber with some constant value you can just do @((NSInteger)sliderValue)

[the remark this answers is not in the document]
Comment (wxie-git) [reply Urban-Fun/View Controllers/FiltersViewController.m:36]: Another way to simplify this is to create a static array of your distances eg
static const NSArray<NSNumber *> *const kDistances = @[@0, @5, @10, ... @(-1)]; // put this above your class implementation

// In the function just do something like this, can simplify your code to 3 lines basically
self.filtersDictionary[@"distance"] = kDistances[(int)sliderValue];
self.filtersDictionary[@"distanceSliderValue"] = @((int)sliderValue];

File: Urban-Fun/Cells/TimelineCell.h [hunk 2]
```diff
@@ -29,6 +29,7 @@ NS_ASSUME_NONNULL_BEGIN
 @property (strong, nonatomic) IBOutlet UIButton *timelineFavoriteButton;
 @property (strong, nonatomic) IBOutlet UILabel *timelineUsernameLabel;
 @property (strong, nonatomic) IBOutlet UIButton *timelineJoinButton;
+@property (strong, nonatomic) IBOutlet NSNumber *distanceDouble;
```
[the remark this answers is not in the document]
Comment (mbooker5) [reply Urban-Fun/Cells/TimelineCell.h:32]: I don't know why but I removed it

File: Urban-Fun/View Controllers/SearchViewController.m [hunk 3]
```diff
@@ -70,91 +74,64 @@ - (void) locationManager:(CLLocationManager *)manager didUpdateLocations:(NSArra
...
             }
-            for (User *user in users) {
-                if (![self.usersArray containsObject:user]){
-                    [self.usersArray addObject:user];
+            if (self.filtersDictionary[@"categoriesCount"] > 0){
+                NSMutableArray *temporaryCategoryArray = [[NSMutableArray alloc] init];
+                for (Activity *activity in self.activitiesArray){
+                    for (Category *category in self.filtersDictionary[@"categories"]){
+                        if ([activity.categories containsObject:category]){
+                            if (![temporaryCategoryArray containsObject:activity]){
```
[the remark this answers is not in the document]
Comment (mbooker5) [reply Urban-Fun/View Controllers/SearchViewController.m:129]: It was working properly but your code works too and is simpler so I'll use it instead. Thanks!

**Pull request merged**